core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -1523,5 +1529,30 @@ private void validateStringCharacterLimit() {
     }
   }
 
-
+  /**
+   * This method validates the DETAIL_QUERY_BATCH_SIZE. If some invalid input is set, we use the
+   * default value for this property
+   */
+  private void validateDetailQueryBatchSize() {
+    String batchSizeString =
+        carbonProperties.getProperty(CarbonCommonConstants.DETAIL_QUERY_BATCH_SIZE);
+    if (batchSizeString == null) {
+      carbonProperties.setProperty(DETAIL_QUERY_BATCH_SIZE,
+          Integer.toString(CarbonCommonConstants.DETAIL_QUERY_BATCH_SIZE_DEFAULT));
+      LOGGER.info("Using default value for Query Batch Size " + DETAIL_QUERY_BATCH_SIZE_DEFAULT);
```

Comment:
instead of "Query Batch size" use "carbon.detail.batch.size". so that user know that it is a carbon property

Reply:
Done.